pkg/apis/pipeline/v1/matrix_types.go

```diff
@@ -277,6 +277,24 @@ func (m *Matrix) HasParams() bool {
 	return m != nil && m.Params != nil && len(m.Params) > 0
 }
 
+// ContainsWholeArrayResultReferences returns true if the Matrix contains
+// references to whole array results [*]
+func (m *Matrix) ContainsWholeArrayResultReferences() bool {
```

Comment:
I know @pritidesai and possibly others have a preference for member functions, but does this need to go in the apis package? It reduces duplication between v1beta1 and v1, and the number of exported functions in apis where we generally at least try to maintain backwards compatibility.

Reply:
@lbernick moved outside the apis pkg